Userland/Libraries/LibJS/Runtime/IteratorOperations.cpp

```diff
@@ -114,10 +124,13 @@ Completion iterator_close(Object& iterator, Completion completion)
 
         // c. Set innerResult to Call(return, iterator).
         auto result_or_error = vm.call(*return_method, &iterator);
-        if (result_or_error.is_error())
+        if (result_or_error.is_error()) {
             inner_result_or_error = result_or_error.release_error();
-        else
+        } else {
             inner_result = result_or_error.release_value();
+            if (is_async == IteratorHint::Async)
```

Comment:
`// d. If innerResult.[[Type]] is normal, set innerResult to Await(innerResult.[[Value]]).`